source/adapters/level_zero/event.cpp

```diff
@@ -882,6 +882,12 @@ urEventRelease(ur_event_handle_t Event ///< [in] handle of the event object
 ) {
   Event->RefCountExternal--;
   UR_CALL(urEventReleaseInternal(Event));
+  // If this is a Completed Event Wait Out Event, then we need to cleanup the
+  // event at user release and not at the time of completion.
+  if (Event && Event->CommandType == UR_COMMAND_EVENTS_WAIT &&
```

Comment:
Event is not taken by a reference by urEventReleaseInternal, so `Event` can't become NULL here. This doesn't protect us from use-after-free.
My primary concern is the event being deleted here:
https://github.com/oneapi-src/unified-runtime/blob/main/source/adapters/level_zero/event.cpp#L1143

Which can occur e.g., with interop events where ownership is not passed over to UR.

Reply:
Updated to save the details on the event before the event release to avoid a use after delete possible scenario.